### ISSUE TITLE
Dashboard should show TaskRun sidecars + their logs

### ISSUE THREAD

**Issue by bitsofinfo** [tektoncd/dashboard#972]
Dashboard should show TaskRun sidecars + their logs

**Comment by AlanGreene** [comment]
related issue for CLI: https://github.com/tektoncd/cli/issues/655

**Comment by a-roberts** [comment]
Yeah, this sounds like a great addition, asked @vdemeester for an example and https://github.com/tektoncd/pipeline/blob/master/examples/taskruns/dind-sidecar.yaml looks suitable for working with

**Comment by AlanGreene** [comment]
I think it would be a good idea to consider this along with https://github.com/tektoncd/dashboard/issues/572 and https://github.com/tektoncd/dashboard/issues/1445 to ensure we provide a good experience and don't introduce too many different mechanisms for accessing errors / logs.

**Comment by AlanGreene** [comment]
This might also benefit from some of the updates being made for https://github.com/tektoncd/dashboard/issues/1775.

**Comment by AlanGreene** [comment]
This should have been linked to https://github.com/tektoncd/dashboard/issues/1445 which we'll need to tackle first, I've just added it as a blocker now.

There are a number of closely related issues that require us to revisit how we structure and display the Task / step / log details so it all works as users expect. You can see some of the discussion in https://github.com/tektoncd/dashboard/issues/1445 and the other linked issues above. Any proposals on this are welcome, and we can discuss them in more detail before implementation.

**Comment by AlanGreene** [comment]
freezing this so it doesn't get auto-closed since this is something we do want to add

/lifecycle frozen

**Comment by wingyplus** [comment]
Thanks for your information. Let me catchup with those issues first. :)

**Comment by shinji62** [comment]
@AlanGreene Any update on this one ? We will need to do that too, and not sure how Tekton Dashboard is going to handle that.

**Comment by wingyplus** [comment]
@shinji62 Currently it's still blocking by #1445. I've worked on color on that issue but I've no time to look it more on that issue at the moment.

**Comment by AlanGreene** [comment]
This is being considered as part of https://github.com/tektoncd/dashboard/issues/2306 which is currently in progress. No ETA yet, but we'll be sharing updates as we start to make progress.

**Comment by sellitforcache** [comment]
Excellent, thanks!

**Comment by AlanGreene** [comment]
This is being added to the PipelineRun details page by #4337 as part of #2306. The new design will be available as an opt-in preview in the next Dashboard release (v0.60).

The changes will be rolled out to the TaskRun page too in a future release, and once stable will replace the existing UI.
